### DISCUSSION TITLE
Preact bindings

### DISCUSSION THREAD

**Discussion by mariosant** [pmndrs/zustand#401] [Ideas] [answered]
I am wondering, would the zustand author be interested in  a PR to include preact bindings?

**Comment by dai-shi** [comment, marked as answer]
Doesn't zustand currently work in preact? or preact/compat?

**Reply by mariosant** [reply]
It should work with preact/compat. I think this layer adds to the bundle though - unless I am mistaken.

**Reply by dai-shi** [reply]
So, you want to use zustand with preact without preact/comat?
I don't think we add such support in zustand core, especially because we are moving ahead to useMutableSource #160 in the near future.
I'm personally interested in how zustand/preact would look like though.
How about creating a 3rd party library that depends on `zustand/vanilla`? We could list it in https://github.com/pmndrs/zustand/wiki/3rd-Party-Libraries.

**Reply by mariosant** [reply]
> So, you want to use zustand with preact without preact/comat?

Indeed @dai-shi.

Thanks, I will try to do so and ping you once I have something up and running.

**Reply by markusbkk** [reply]
Any update on this, @mariosant ?
I'd be interested in this too (although I'm new to preact and haven't looked into their own state management mechanisms, yet).

**Comment by saolof** [comment]
Deepsignal could also be a good option for Zustand-like store that is idiomatic to preact, being built on top of preact/signals which has direct integration with the preact VDOM. All the reactivity is provided by signals and deepsignal itself only exposes two functions to make nested structs of them.

**Reply by dai-shi** [reply]
Unrelated, but fyi. Here's my take inspired by preactjs/signals-react: https://github.com/dai-shi/zustand-signal

**Reply by saolof** [reply]
@dai-shi  That looks very interesting. It integrates with reacts create element? Would it work with preact/compat?

**Reply by dai-shi** [reply]
It hacks React's createElement. I don't know if it works with preact/compat.

**Comment by mindplay-dk** [comment]
I know this is an old thread, but someone built a Zustand adapter for Preact. 😄

https://github.com/aterana3/preact-zustand

It's a very small and simple package, and it has test coverage - I gave it a try, and it works.

@dai-shi might be worth documenting somewhere for others to find? 🙂

**Reply by dai-shi** [reply]
Sure, the author or anyone can send a PR to add it to https://github.com/pmndrs/zustand/blob/main/docs/integrations/third-party-libraries.md.

On a related note, I develop https://github.com/zustandjs/use-zustand which doesn't use `useSyncExternalStore`, so it may work with preact too.

**Reply by mindplay-dk** [reply]
I should have said, the idea of that library is to get vanilla Zustand to work *without* `preact-compat` - I don't think there's any problems using Zustand with `preact-compat`, afaik? 😊

**Reply by dai-shi** [reply]
Right, as preact/compat should have useSyncExternlStore now.
use-zustand may work without preact/compat if a bundler aliases the react import (to preact/hooks not compat.)
